From b1dacdd626501dd2e86f7d6d0f49a7df177c1797 Mon Sep 17 00:00:00 2001
From: Olivier Duchateau <duchateau.olivier@gmail.com>
Date: Mon, 1 Jul 2019 19:42:05 +0200
Subject: [PATCH] More spacing between GtkGrid, GtkInfoBar, and GtkBox widgets

---
 src/xfce4-screensaver-dialog.ui | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/src/xfce4-screensaver-dialog.ui b/src/xfce4-screensaver-dialog.ui
index 53f5caa..443d1ae 100644
--- a/src/xfce4-screensaver-dialog.ui
+++ b/src/xfce4-screensaver-dialog.ui
@@ -99,9 +99,10 @@
               <object class="GtkGrid">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
-                <property name="margin_left">24</property>
-                <property name="margin_right">24</property>
-                <property name="margin_top">24</property>
+                <property name="margin_left">14</property>
+                <property name="margin_right">14</property>
+                <property name="margin_top">14</property>
+                <property name="margin_bottom">8</property>
                 <property name="row_spacing">6</property>
                 <property name="row_homogeneous">True</property>
                 <child>
@@ -315,9 +316,10 @@
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="halign">end</property>
-                <property name="margin_left">24</property>
-                <property name="margin_right">24</property>
-                <property name="margin_bottom">24</property>
+                <property name="margin_left">8</property>
+                <property name="margin_right">8</property>
+                <property name="margin_top">14</property>
+                <property name="margin_bottom">14</property>
                 <property name="spacing">6</property>
                 <property name="homogeneous">True</property>
                 <child>
-- 
2.21.0
