From 70c470b089b02da5e344a0a200acb3f66a388364 Mon Sep 17 00:00:00 2001
From: Andre Miranda <andreldm@xfce.org>
Date: Sun, 25 Aug 2019 18:37:51 -0300
Subject: fallback style: Highlight selected icons (Bug #15866)

---
 src/xfdesktop-application.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/xfdesktop-application.c b/src/xfdesktop-application.c
index 539266a8..b655ca32 100644
--- a/src/xfdesktop-application.c
+++ b/src/xfdesktop-application.c
@@ -150,10 +150,16 @@ const gchar *fallback_CSS =
 "XfdesktopIconView.view .label {"
 "	text-shadow: 1px 1px 2px black;"
 "}"
+"XfdesktopIconView.view .label:active {"
+"	color: @theme_selected_fg_color;"
+"}"
 "XfdesktopIconView .rubberband {"
 "	background: alpha(@theme_selected_bg_color, 0.2);"
 "	border: 1px solid @theme_selected_bg_color;"
 "	border-radius: 0;"
+"}"
+"XfdesktopIconView:active {"
+"	color: @theme_selected_bg_color;"
 "}";
 
 G_DEFINE_TYPE(XfdesktopApplication, xfdesktop_application, G_TYPE_APPLICATION)
-- 
cgit v1.2.1

