From ed4ae8f7a3c76f1b8d8dd78d590cfe128bd4ad95 Mon Sep 17 00:00:00 2001
From: Simon Steinbeiss <simon.steinbeiss@elfenbeinturm.at>
Date: Tue, 6 Aug 2019 17:14:32 +0200
Subject: Improve size of critical power level spin

---
 data/interfaces/xfpm-settings.ui | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/interfaces/xfpm-settings.ui b/data/interfaces/xfpm-settings.ui
index 62e5217..4bc8597 100644
--- a/data/interfaces/xfpm-settings.ui
+++ b/data/interfaces/xfpm-settings.ui
@@ -968,6 +968,7 @@
                                     <property name="primary_icon_activatable">False</property>
                                     <property name="secondary_icon_activatable">False</property>
                                     <property name="adjustment">critical-power-level-spin-adjustment</property>
+                                    <property name="width-chars">6</property>
                                     <property name="width-request">100</property>
                                     <signal name="output" handler="critical_spin_output_cb" swapped="no"/>
                                     <signal name="value-changed" handler="critical_level_value_changed_cb" swapped="no"/>
-- 
cgit v1.2.1

