From d4e7075401b6ef60cf1f015a488cf4b573f9b8dd Mon Sep 17 00:00:00 2001
From: Yves-Alexis Perez <corsac@debian.org>
Date: Tue, 20 Aug 2019 17:20:09 +0200
Subject: prepend $XDG_DATA_HOME/thumbnailers/ to thumbnailers directory (Bug
 #15858)

---
 plugins/desktop-thumbnailer/desktop-thumbnailer-provider.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plugins/desktop-thumbnailer/desktop-thumbnailer-provider.c b/plugins/desktop-thumbnailer/desktop-thumbnailer-provider.c
index 9340822..1054adc 100644
--- a/plugins/desktop-thumbnailer/desktop-thumbnailer-provider.c
+++ b/plugins/desktop-thumbnailer/desktop-thumbnailer-provider.c
@@ -236,6 +236,11 @@ desktop_thumbnailer_provider_get_thumbnailers (TumblerThumbnailerProvider *provi
 
   uri_schemes = tumbler_util_get_supported_uri_schemes ();
 
+  /* prepend $XDG_DATA_HOME/thumbnailers/ to the directory list */
+  dirname = g_build_filename (g_get_user_data_dir (), "thumbnailers", NULL);
+  directories = g_list_prepend (directories, g_file_new_for_path (dirname));
+  g_free (dirname);
+
   /* build $XDG_DATA_DIRS/thumbnailers dirnames and prepend them to the list */
   data_dirs = g_get_system_data_dirs ();
 
-- 
cgit v1.2.1

